client/state/stats/lists/actions.js

```diff
@@ -66,7 +66,6 @@ export function requestSiteStats( siteId, statType, query ) {
 		let apiNamespace;
 		if ( wpcomV1Endpoints.hasOwnProperty( statType ) ) {
 			subpath = wpcomV1Endpoints[ statType ];
-			apiNamespace = 'rest/v1.1';
```

Comment:
I told you so! 😄 Apparently it was not just a style nit, but the mere presence of `apiNamespace` means we are in WP REST API territory and that the envelope query param is different (WP.com `http_envelope` vs WP `_envelope`).

Reply:
I stand corrected! I'll just admit and pretend this makes sense in the first place 😆

Reply:
Interesting, what was the original rationale for adding this line, and why does removing it fixes the envelope query param (as described in the PR's title)?

Reply:
Good question @fullofcaffeine, let me add some context.

So, this line was added in #57518 as part of our work to refactor away from `wpcom.undocumented()` methods. 

As you can see, in [the old code](https://github.com/Automattic/wp-calypso/pull/57518/files#diff-c3e509bb1fa8af9056af182e044d339272e486c7911f0fa3bb82695b8b211654L53), we didn't specify `apiNamespace` specifically. 

However, in the new version, we did. And if you take a look at the library that performs the requests, we'll notice that the HTTP envelope argument [depends on the presence of `isRestApi`](https://github.com/Automattic/wp-calypso/blob/trunk/packages/wpcom-xhr-request/src/index.js#L266), which [itself depends on](https://github.com/Automattic/wp-calypso/blob/trunk/packages/wpcom-xhr-request/src/index.js#L216) whether `apiNamespace` has been **defined**. So by defining `apiNamespace`, we were no longer using the HTTP envelope. 

Now, you might ask, why does the HTTP envelope change anything? Well, it alters the behavior of the endpoint response in a pretty significant way. It will change the endpoint to always send HTTP 200 (regardless of the outcome) and will wrap the JSON output in a wrapper containing the "real" HTTP status code and some of the headers. This was useful historically because many clients could potentially block non-200 status code responses. 

So, to summarize, in our case, the client was expecting the response to be in the format that's provided by an HTTP-envelope-wrapped endpoint, and by declaring `apiNamespace`, we were technically removing the envelope, rendering the output in a format that's unexpected by the client.

Let me know if that makes sense 😉

Reply:
> So by defining `apiNamespace`, we were no longer using the HTTP envelope.

More precisely, we were using wrong query param to request the envelope. The old v1 API uses `http_envelope`, the new v2 uses `_envelope`. When `apiNamespace` is specified, the library assumes v2, even when we passed a v1 namespace string.